test/e2e/validation_test.go

```diff
@@ -165,37 +155,50 @@ func validateInstanceSecret(kubeclient kubernetes.Interface, namespace, instance
 			return false, err
 		}
 		encodedCreds := instanceSecret.Data[instanceID]
-
-		var creds tracker.Credentials
 		if err := json.Unmarshal(encodedCreds, &creds); err != nil {
 			return false, errors.Wrap(err, "unmarshalling creds failed")
 		}
-
-		if err := validateConnectivity(ipAddress, instanceID, creds); err == nil {
-			log.Println("Successfully validated the SSH Connection")
-			return true, nil
-		}
-
-		log.Printf("failed with error for creds %v: %v", creds, err)
-		return false, nil
+		return true, nil
 	})
+	return creds, err
+}
+
+// validateInstanceSecret validates the instance secret.
+func (t *testContext) validateInstanceSecret(instanceID string) error {
+	ipAddress, err := t.getInstanceIP(instanceID)
+	if err != nil {
+		return err
+	}
+	creds, err := t.getCredsSecretData(instanceID)
+	if err != nil {
+		return err
+	}
+	if creds == (tracker.Credentials{}) {
+		return errors.New("expected credentials to be present but got a nil value")
+	}
+	windowsVM, err := t.getWindowsVM(ipAddress, instanceID, creds)
+	if err != nil {
+		return err
+	}
+	t.windowsVMs = append(t.windowsVMs, windowsVM)
```

Comment:
I still don't follow how the same test context can be used in different tests so that `t.windowsVMs` can be reused to run tests inside the VMs. Are you saying that every time we want to run tests against a WindowsVM, we have to follow the pattern introduced in `testValidateSecrets()`?

Reply:
> Are you saying that every time we want to run tests against a WindowsVM, we have to follow the pattern introduced in testValidateSecrets()?

That is right, since we've testContext per every test, we cannot share it across multiple tests but we need to generate it for every test. I was initially thinking of sharing the context across the entire test suite but it seems to digress from sub-test pattern. Another reason was the way operator-sdk's testFramework struct was structured. They've an inverted dependency on `testing.T` and that's why it has to unique for every sub-test. Since we composed framework
's testFramework struct, we have to be unique for every test too. The other approach is not to rely on operator-sdk's testFramework which might it hard for us atm.